dotCMS/src/main/java/com/dotmarketing/portlets/cmsmaintenance/factories/CMSMaintenanceFactory.java

```diff
@@ -17,138 +18,175 @@
 import java.util.Map;
 
 /**
- * This class is use to fix identifiers inconsistencies in db,
- * solving the problem of identifiers without a working or live asset
- * @author Oswaldo
+ * This tool allows CMS users to delete entries of versionable dotCMS objects
+ * that are older than a specified date. These versions can be accessed through
+ * the <b>History</b> tab of such objects. The list of elements that are
+ * versionable in the system are:
+ * <ul>
+ * <li>Contentlets - including HTML Pages as Content, Files as Content, Sites,
+ * etc.</li>
+ * <li>Containers.</li>
+ * <li>Templates.</li>
+ * <li>Links.</li>
+ * <li>Workflow History.</li>
+ * </ul>
  *
+ * @author root
+ * @since Mar 22, 2012
  */
 public class CMSMaintenanceFactory {
 
-	public static int deleteOldAssetVersions (Date assetsOlderThan) {
-		int counter = 0;
-		int auxCount = 0;
-
+    /**
+     * Deletes dotCMS object versions that are older than the specified date. Such a
+     * date is compared to the date in which a version of an object was created. The
+     * list of objects may include:
+     * <ul>
+     * <li>Contentlets - including HTML Pages as Content, Files as Content, Sites,
+     * etc.</li>
+     * <li>Containers.</li>
+     * <li>Templates.</li>
+     * <li>Links.</li>
+     * <li>Workflow History.</li>
+     * </ul>
+     *
+     * @param assetsOlderThan
+     *            The user-specified date up to which object versions will be
+     *            deleted.
+     * @return The total number of versions that were deleted.
+     */
+    public static int deleteOldAssetVersions(final Date assetsOlderThan) {
+        Logger.info(CMSMaintenanceFactory.class, "------------------------------------------------------------");
+        Logger.info(CMSMaintenanceFactory.class, " Executing the Drop Old Assets Versions tool");
+        Calendar startDate = Calendar.getInstance();
+        startDate.setTime(assetsOlderThan);
+        startDate.add(Calendar.YEAR, -2);
+        startDate.set(Calendar.HOUR_OF_DAY, 0);
+        startDate.set(Calendar.MINUTE, 0);
+        startDate.set(Calendar.SECOND, 0);
+        startDate.set(Calendar.MILLISECOND, 0);
+        final Calendar oldestInodeDate = getOldestInodeDate();
+        startDate = (null != oldestInodeDate ? oldestInodeDate : startDate);
+        Logger.info(CMSMaintenanceFactory.class, String.format("- Deleting versions older than '%s'",
+                UtilMethods.dateToHTMLDate(assetsOlderThan, "yyyy-MM-dd")));
+        Logger.info(CMSMaintenanceFactory.class, " ");
+        int totalRecords = 0;
+        while (startDate.getTime().before(assetsOlderThan) || startDate.getTime().equals(assetsOlderThan)) {
+            try {
+                HibernateUtil.startTransaction();
+                int deletedRecords = removeOldVersions(startDate);

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'deletedRecords' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5534906)

Reply:
Fixed.